src/ngraph/onnx_import/graph.cpp

```diff
@@ -0,0 +1,46 @@
+/*******************************************************************************
+ * Copyright 2017-2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#include "graph.hpp"
+#include "node.hpp"
+
+using namespace ngraph::onnx_import;
+
+Graph::Graph(const onnx::GraphProto& graph_proto)
+    : m_graph_proto(graph_proto)
+{
+    for (const auto& node : m_graph_proto.node())
+        m_nodes.emplace_back(Node(node, this));
```

Comment:
Correct use of emplace_back is `m_nodes.emplace_back(node, this)`, the object will be constructed in place. In your approach the object will be constructed before emplace_back is called.